devices/wholeBodyDynamics/WholeBodyDynamicsDevice.cpp

```diff
@@ -1428,17 +1619,40 @@ void WholeBodyDynamicsDevice::readContactPoints()
         {
             if (contactsReadFromSkin.empty())
             {
-
-                for(size_t subModel = 0; subModel < nrOfSubModels; subModel++)
+                if(!overrideContactFramesSelected)
                 {
-                    bool ok = measuredContactLocations.addNewContactInFrame(estimator.model(),
-                                                                            subModelIndex2DefaultContact[subModel], //frameIndex in iDynTree
-                                                                            iDynTree::UnknownWrenchContact(iDynTree::FULL_WRENCH,iDynTree::Position::Zero()));
-                    if( !ok )
+                    for(size_t subModel = 0; subModel < nrOfSubModels; subModel++)
+                        {
+                        bool ok = measuredContactLocations.addNewContactInFrame(estimator.model(),
+                                                                                subModelIndex2DefaultContact[subModel], //frameIndex in iDynTree
+                                                                                iDynTree::UnknownWrenchContact(iDynTree::FULL_WRENCH,iDynTree::Position::Zero()));
+                        if( !ok )
+                        {
+                            yWarning() << "wholeBodyDynamics: Failing in adding default contact for submodel " << subModel;
+                        }
+                    }
+                }
+                else
+                {
+                    for(size_t subModel = 0; subModel < nrOfSubModels; subModel++)
                     {
-                        yWarning() << "wholeBodyDynamics: Failing in adding default contact for submodel " << subModel;
+                        //iterate for each added contact frame
+                        for(size_t ovFrame = 0; ovFrame < overrideContactFrames.size(); ovFrame++)
```

Comment:
can these multiple usages of similar for loops be replaced by a function?

Reply:
After reducing the nested for-loops to one for-loop (in https://github.com/robotology/whole-body-estimators/pull/48/commits/38c21cdf2a12d33f0fd9a3f736d69353bb89e86e), it doesn't feel necessary to put it in a function. what do you think?